programs/rodeos/wasm_ql_plugin.cpp

```diff
@@ -17,14 +17,45 @@ namespace b1 {
 
 struct wasm_ql_plugin_impl : std::enable_shared_from_this<wasm_ql_plugin_impl> {
    bool                                         stopping     = false;
+   uint32_t                                     max_retries  = 0xffff'ffff;
+   uint32_t                                     retried      = 0;
    std::shared_ptr<const wasm_ql::http_config>  http_config  = {};
    std::shared_ptr<const wasm_ql::shared_state> shared_state = {};
    std::shared_ptr<wasm_ql::http_server>        http_server  = {};
+   boost::asio::deadline_timer                  timer;
 
-   void start_http() { http_server = wasm_ql::http_server::create(http_config, shared_state); }
+   wasm_ql_plugin_impl() : timer(app().get_io_service()) {}
+
+   void start_http() {
+      http_server = wasm_ql::http_server::create(http_config, shared_state);
+      if (!http_server)
+         schedule_retry();
+   }
+
+   void schedule_retry() {
+      if (retried++ < max_retries) {
+         timer.expires_from_now(boost::posix_time::seconds(1));
+         timer.async_wait([this](auto&) {
+            ilog("retry...");
+            try {
+               try {
+                  start_http();
+               }
+               FC_LOG_AND_RETHROW()
+            } catch (...) {
+               elog("shutting down");
+               app().shutdown();
```

Comment:
Directly calling `app().shutdown()` is dangerous.  It's better to call `app().quit()` and let appbase break out of the event loop normally.

Reply:
We really should make that method private.

Reply:
Yep. And switch http_plugin to using quit.

Reply:
I saw that. I'll create a PR to develop for it.